code/game/sound_loop.dm

```diff
@@ -0,0 +1,127 @@
+/*
+//	Not works properly
+*/
+
+datum/sound_loop
+	var/list/listeners = list()
+	var/atom/source
+
+	var/file
+	var/range = 3
+	var/volume = 100
+	var/frequency = 0
+	var/falloff = 0
+
+	var/playing = FALSE
+
+	var/force_update = FALSE
+
+
+datum/sound_loop/New(var/atom/source, soundname, vol as num, range as num, freq = 0, falloff = FALLOFF_SOUNDS)
+	src.source = source
+	src.file = soundname
+	src.volume = vol
+	src.range = range
+	src.frequency = freq
+	src.falloff = falloff
+
+datum/sound_loop/Destroy()
+	for(var/mob/M in listeners)
+		remove_listener(M)
+
+datum/sound_loop/proc/play()
+	playing = TRUE
+	src.force_update = TRUE
+
+
+datum/sound_loop/proc/stop()
+	playing = FALSE
+	src.force_update = TRUE
+
+
+datum/sound_loop/proc/set_volume(var/newvol)
```

Comment:
Please use "snake-case" variables style and avoid abbreviations (var/new_volume should be nice). Same comments for other datum methods.